### PR TITLE
fix(cid-as-string): Return CID as string instead of byte array #787

### LINKED ISSUE #787: Message RPC returns IPFS CIDv1s as String base32
Our output of IPFS CIDs is not properly sanitized and serialized.
Outcome: only output IPFS CIDv1 in `base32` encoding

Partially dependent on #786

Breaking Change:
- Before: `MessageResponse` returned CID of type `Option<Vec<u8>>` which could have been anything
- After: `MessageResponse` returns CID of type `Option<string>` which is the cidv1 in `base32`

Related:
- #738 
- #786 
- CID Spec: https://github.com/multiformats/cid
- CID Inspector: https://cid.ipfs.tech
- JS for multiformats https://github.com/multiformats/js-multiformats/blob/master/src/cid.js

## Acceptance Criteria
- [x] `add_ipfs_message` with:
    - [x] string: `bafybeihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku`
    - [x] stores payload as hex: `0x01701220e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`
    - [x] Returns `get_messages_by_schema_id` base32 string of:`bafybeihdwdcefgh4dqkjv67uzcmw7ojee6xedzdetojuzjevtenxquvyku`
- [x] Better documentation around the IPFS response as always `CIDv1`
    - [x] https://libertydsnp.github.io/frequency/common_primitives/messages/struct.MessageResponse.html#structfield.cid

### PR DESCRIPTION
**Pull request by JoeCap08055**
# Goal
The goal of this PR is to return CID as a string instead of `Vec<u8>` bytes. Based on:

> Peter Frank
  [6:36 PM](https://unfinished-team.slack.com/archives/C01L4J5SQA1/p1675467404682619)
@here think I found a bug, or really an incomplete ticket. 
[@Aziz Abada](https://unfinished-team.slack.com/team/U02J2JHFZJ4)
 is integrating against the messages.getMessagesBySchemaId  call and for the cid itself we are being returned a byte array (really a JSON array of very small numbers that are less than u8 :)). It appears this was noticed in this ticket https://github.com/LibertyDSNP/frequency/issues/787 and the comment on this code https://github.com/LibertyDSNP/frequency/blob/main/common/primitives/src/messages.rs#L38-L39 talks about the base32 encoding in the comment but it doesn’t actually change the data type from Option<Vec<u8>> to Option<string>. Just wanted to call this out, we will be adding a method to convert this on our side for now receiving a naked byte array (but especially one that is really a string) isn’t terribly useful but not sure how we go about fixing/changing this.

Closes #787 

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [ ] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

**Review by enddynayn — APPROVED** [review]
LGTM 👍

File: integration-tests/messages/addIPFSMessage.test.ts [hunk 1]
```diff
@@ -106,8 +108,9 @@ describe("Add Offchain Message", function () {
     });
 
     it("should successfully retrieve added message and returned CID should have Base32 encoding", async function () {
-        const f = await firstValueFrom(ExtrinsicHelper.api.rpc.messages.getBySchemaId(schemaId, { from_block: 0, from_index: 0, to_block: 999, page_size: 999 }));
-        const response: MessageResponse = f.content[0];
+        const f = await firstValueFrom(ExtrinsicHelper.api.rpc.messages.getBySchemaId(schemaId, { from_block: starting_block, from_index: 0, to_block: starting_block + 999, page_size: 999 }));
+        const response: MessageResponse = f.content[f.content.length - 1];
+        console.log(`Got messages: schema ${schemaId}, block: ${response.block_number}`);
```
Comment (shannonwells) [inline integration-tests/messages/addIPFSMessage.test.ts:113]: extraneous console log

**Review by shannonwells — APPROVED** [review]
Looks ok - just the leftover comment

**Review by saraswatpuneet — APPROVED** [review]

File: integration-tests/messages/addIPFSMessage.test.ts (hunk 1 above)
Comment (JoeCap08055) [reply integration-tests/messages/addIPFSMessage.test.ts]: Removed, thanks for catching it

**Pull request merged**